noImplicitReturns does not work correctly in class constructors in nightlies

**TypeScript Version:** 

master

**Code**

```ts
class C {
  constructor() {
    return;
  }
}
```

**Expected behavior:**

```
$ node built/local/tsc.js index.ts --noImplicitReturns
```

**Actual behavior:** 

```
$ node built/local/tsc.js index.ts --noImplicitReturns
index.ts(3,5): error TS7030: Not all code paths return a value.
```

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -13965,7 +13965,7 @@ namespace ts {
                         }
                     }
                 }
-                else if (compilerOptions.noImplicitReturns && !isUnwrappedReturnTypeVoidOrAny(func, returnType)) {
+                else if (func.kind !== SyntaxKind.Constructor && compilerOptions.noImplicitReturns && !isUnwrappedReturnTypeVoidOrAny(func, returnType)) {
                     // The function has a return type, but the return statement doesn't have an expression.
                     error(node, Diagnostics.Not_all_code_paths_return_a_value);
                 }

--- a/tests/baselines/reference/noImplicitReturnInConstructors.js
+++ b/tests/baselines/reference/noImplicitReturnInConstructors.js
@@ -1,0 +1,14 @@
+//// [noImplicitReturnInConstructors.ts]
+class C {
+  constructor() {
+    return;
+  }
+}
+
+//// [noImplicitReturnInConstructors.js]
+var C = (function () {
+    function C() {
+        return;
+    }
+    return C;
+}());

--- a/tests/baselines/reference/noImplicitReturnInConstructors.symbols
+++ b/tests/baselines/reference/noImplicitReturnInConstructors.symbols
@@ -1,0 +1,8 @@
+=== tests/cases/compiler/noImplicitReturnInConstructors.ts ===
+class C {
+>C : Symbol(C, Decl(noImplicitReturnInConstructors.ts, 0, 0))
+
+  constructor() {
+    return;
+  }
+}

--- a/tests/baselines/reference/noImplicitReturnInConstructors.types
+++ b/tests/baselines/reference/noImplicitReturnInConstructors.types
@@ -1,0 +1,8 @@
+=== tests/cases/compiler/noImplicitReturnInConstructors.ts ===
+class C {
+>C : C
+
+  constructor() {
+    return;
+  }
+}

--- a/tests/cases/compiler/noImplicitReturnInConstructors.ts
+++ b/tests/cases/compiler/noImplicitReturnInConstructors.ts
@@ -1,0 +1,6 @@
+// @noImplicitReturns: true
+class C {
+  constructor() {
+    return;
+  }
+}

